### PR TITLE
Add clang-format file and clang-format target

### LINKED ISSUE #7: Missing includes in OculusModule.hpp
When I try to build the application, I get the following error:
```
In file included from /home/giulio/robot-code/walking-teleoperation/modules/Oculus_module/src/main.cpp:14:0:
/home/giulio/robot-code/walking-teleoperation/modules/Oculus_module/include/OculusModule.hpp:102:12: error: ‘RpcClient’ in namespace ‘yarp::os’ does not name a type yarp::os::RpcClient m_Joyrpc;
```

The bug can be easily fixed adding the following line in `OculusModule.hpp`
```cpp
#include < yarp/os/RpcClient.h>
```

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
This PR wants to:
1. Fix #7 adding the missing include 
2. Add a `.clang-format` file (from https://raw.githubusercontent.com/robotology-playground/clang-format/1811db969fc5adff4b5cd5a5d9c087aadd9265e0/clang-format) and a `clang-target`CMake target (from https://github.com/robotology/sdf-modelica/blob/master/cmake/AddClangFormatTarget.cmake)  to automatically format with `clang-format` the source code. 

The `.clang-format` file was tested witch `clang-format` version 6.0.0

@kouroshD @traversaro

### CODE REVIEW

File: modules/Oculus_module/include/#HandRetargeting.hpp# [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/**
```
Comment (traversaro) [inline modules/Oculus_module/include/#HandRetargeting.hpp#]: `.hpp#` ?

File: cmake/AddClangFormatTarget.cmake [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# https://www.gnu.org/licenses/lgpl-3.0.html
+# or the GNU Lesser General Public License v2.1 :
+# https://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+# at your option.
+
+# Targets to run and check source code with clang-format
+# Inspired from https://gitlab.cern.ch/unige-fei4tel/proteus/commit/8d906a45801c03832531e243f41f5f5a83177de0
+
+# Adding clang-format check and formatter if found
+find_program(CLANG_FORMAT "clang-format-6.0")
```
Comment (traversaro) [inline cmake/AddClangFormatTarget.cmake]: There is any specific reason for which you do not want to match also "clang-format"?

File: modules/Oculus_module/include/#HandRetargeting.hpp# (hunk 1 above)
Comment (GiulioRomualdi) [reply modules/Oculus_module/include/#HandRetargeting.hpp#]: File autogenerated by emacs. Removed in https://github.com/robotology/walking-teleoperation/pull/8/commits/5be53b54d032767ef445450249062c4fae0eee04

File: cmake/AddClangFormatTarget.cmake (hunk 2 above)
Comment (traversaro) [reply cmake/AddClangFormatTarget.cmake]: Ah, I got it. Yes your suggestion is definitely better for forward compatibility.

**Comment by GiulioRomualdi** [comment]
Before merging If you want I may squash all the commits into one. Let me know what do you prefer.

**Comment by traversaro** [comment]
> Before merging If you want I may squash all the commits into one. Let me know what do you prefer.

You are the mantainer, so you decide. : )

**Comment by GiulioRomualdi** [comment]
Emh I'm not the maintainer. I think @mebbaid is.

**Comment by mebbaid** [comment]
> Before merging If you want I may squash all the commits into one. Let me know what do you prefer.

Just catching up to the emails. I can merge as it is and add you as a collaborator , so you will be able to do stuff in the future.

**Pull request merged**